corehq/apps/es/users.py

```diff
@@ -64,13 +64,17 @@ def show_only_inactive(self):
         return query.is_active(False)
 
 
-def domain(domain, allow_enterprise=False):
+def domain(domain, allow_enterprise=False, allow_enterprise_controlled_domains=False):
```

Comment:
Would it be unreasonable to move these arguments and their logic to the places where they're used?  In my opinion, the `domain` filter shouldn't know anything about enterprise permissions, it just accepts a domain or a list of domains, and restricts results to users on those domains.

Then if a report wants to return users on other domains, it can assemble the appropriate list of domains and pass that in.  If these end up being used in a bunch of places, it can still be pulled in to a helper function without loss of readability like so:

```python
return (UserES()
        .domain(get_enterprise_controlled_domains(domain))
        .search_string_query(query, default_fields=search_fields))
```

Reply:
Probably not unreasonable. Quickly grepping suggests [this](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/filters/users.py#L351) is the only place we pass `allow_enterprise=True` to `domain`.

Reply:
agreed that sounds good I'll work on that fix

Reply:
It would be really helpful to add a small note about how setting `allow_enterprise` and `allow_enterprise_controlled_domains` changes the valid domains. For someone trying to understand the report and not aware of Enterprise model, this would get really confusing.